doc/platform_json_enhancement.md

```diff
@@ -0,0 +1,127 @@
+# Platform capability file enhancement #
+
+### Rev 0.1
+
+## Table of Content
+
+## Revision
+
+ | Rev |     Date    |            Author            | Change Description   |
+ |:---:|:-----------:|:----------------------------:|----------------------|
+ | 0.1 |             | Arun Saravanan Balachandran  | Initial version      |
+
+## Scope
+
+This document provides information on the enhancements for platform capability file `platform.json`.
+
+## Definitions/Abbreviations
+
+| Definitions/Abbreviation | Description |
+|--------------------------|-------------|
+| BMC | Baseband Management Controller |
+| NOS | Network Operating System |
+| PSU | Power Supply Unit |
+
+## Overview
+
+Each networking switch has a set of platform components (e.g: Fans, PSUs, LEDs, etc.) and these components in each platform can have different characteristics (like supported colors for a LED). In a given platform, the components could be controlled by a dedicated platform controller (like BMC) or the NOS running on the CPU is required to control it and in the former the control of the components from the NOS could be limited.
+
+In SONiC the platform components' supported attributes are made available via Platform API, but certain platform specific capabilties for the components are not available for the applications.
+
+This document provides the enhancement for `platform.json` to address the above issue.
+
+## Design
+
+Currently, `platform.json` is used for providing the expected structure of the platform components and interface details for supporting dynamic port breakout.
+
+### Platform capabilities field
+
+A new set of `capabilities` fields are introduced in platform.json, for providing platform specific capablities on control and characteristics of the components.
+
+For each component's attribute, the defined `capabilities` fields are as follows:
+
+- "controllable" : A boolean, 'true' if the given attribute can be controlled from the NOS, 'false' otherwise.
```

Comment:
Can you elaborate in this context what controllable from the NOS mean? Most of these attributes would be managed by a daemon, either directly or via vendor code. So if say you change the fan speed, the thermalctld would change it back, and you'd have to stop the container in order to change it. Is this deemed controllable?

Reply:
Can you maybe reword to:

`- "controllable" : Boolean, 'true' if the given attribute can be controlled from the NOS, 'false' if it cannot. Defaults to 'true.'`

So you don't need the note below.

Reply:
> Can you elaborate in this context what controllable from the NOS mean? Most of these attributes would be managed by a daemon, either directly or via vendor code. So if say you change the fan speed, the thermalctld would change it back, and you'd have to stop the container in order to change it. Is this deemed controllable?

If a given platform component can be controlled from the NOS (software-controlled), it is deemed to be 'controllable' and 
if a platform component is controlled by a dedicated platform controller (like BMC) and is not controllable from the NOS running on the CPU, it is deemed as 'not controllable'.